addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,899 @@
+                    }
+
+                    onConnectable(socketChannel);
+                    ByteBuffer buffer = onReadable(socketChannel,
+                            ((BigDecimal) getConfig().get(BUFFER_SIZE)).intValue());
+                    if (buffer != null && buffer.remaining() > 0) {
+                        onRead(buffer, socketChannel);
+                    }
+                }
+
+                // onAcceptable(listenerChannel);
+
+            } catch (IOException e) {
+                logger.trace("An exception occurred while polling the transceiver : '{}'", e.getMessage(), e);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        }
+    };
+
+    public boolean registerTransceiverStatusListener(TransceiverStatusListener transceiverStatusListener) {
+        Objects.requireNonNull(transceiverStatusListener, "It's not allowed to pass a null BlasterStatusListener.");
+
+        boolean result = transceiverStatusListeners.add(transceiverStatusListener);
+        if (result) {
+            onUpdate();
+        }
+        return result;
+    }
+
+    public boolean unregisterTransceiverStatusListener(TransceiverStatusListener transceiverStatusListener) {
+        boolean result = transceiverStatusListeners.remove(transceiverStatusListener);
+        if (result) {
+            onUpdate();
+        }
+        return result;
+    }
+
+    @Override
+    public void onBridgeDisconnected(EthernetBridgeHandler bridge) {
+        // Nothing to do here
+    }
+
+    @Override
+    public void onBridgeConnected(EthernetBridgeHandler bridge) {
+        // Nothing to do here
+    }
+
+    @Override
+    public void onCommandReceived(EthernetBridgeHandler bridge, IrCommand command) {
+
+        logger.debug("Received infrared command '{},{}' for thing '{}'", command.getRemote(), command.getCommand(),
+                this.getThing().getUID());
+
+        for (Channel channel : getThing().getChannels()) {
+            Configuration channelConfiguration = channel.getConfiguration();
+
+            if (channel.getAcceptedItemType().equals(IRtransBindingConstants.RECEIVER_CHANNEL_TYPE)) {
+                IrCommand thingCompatibleCommand = new IrCommand();
+                thingCompatibleCommand.setRemote((String) channelConfiguration.get(IRtransBindingConstants.REMOTE));
+                thingCompatibleCommand.setCommand((String) channelConfiguration.get(IRtransBindingConstants.COMMAND));
+
+                if (command.matches(thingCompatibleCommand)) {
+                    StringType stringType = new StringType(command.getRemote() + "," + command.getCommand());
+                    logger.debug("Received a matching infrared command '{}' for channel '{}'", stringType,
+                            channel.getUID());
+                    updateState(channel.getUID(), stringType);
+                }
+            }
+        }
+    }
+}
```

Comment:
line 900 is missing ;-)
No seriously: This class is incredibly long and hard to understand what happens where. Check if you can split it up into different classes.